spec/controllers/application_controller_spec.rb

```diff
@@ -28,4 +28,20 @@
       expect(user.library_id).to eq '12345'
     end
   end
+
+  describe '#ldap_attributes' do
+    it 'uses the request env' do
+      allow(controller.request).to receive(:env).and_return('a' => 1)
+      expect(controller.send(:ldap_attributes)).to include 'a' => 1
+    end
+
+    it 'enriches the request env with local settings' do
+      allow(controller).to receive_messages(user_id: 'some-user')
+      allow(controller.request).to receive(:env).and_return('a' => 1)
+      allow(Rails.env).to receive(:development?).and_return(true)
+      Settings.fake_ldap_attributes ||= {}
+      allow(Settings).to receive(:fake_ldap_attributes).and_return('some-user' => { 'b' => 2 })
```

Comment:
This kinda surprised me a bit. Can you tell me more about what's going on here w/ this test (stubbing `#development?`, setting a rails_config setting if not set, then stubbing it).  I'm guessing there is a logical reason to do this, I'm just trying to figure out what it is.